### ISSUE TITLE
regression in I.A. ?

### ISSUE THREAD

**Issue by a-detiste** [Andrettin/Wyrmsun#44]
I'm now at quests "Gathering Materials" & "Hills of Shorbear", the enemy seams only to want to gather ressources; never attack, it only attack in a automated way if it's been attacked first. It won't produce fighter units either ?!

I tried some other previous quest "Northwards to the sea" & i have the same behaviour: after the first batch of fighters (like an handfull), enemy doesn't produce soldiers anymore.

**Comment by Andrettin** [comment]
Strange, I didn't change anything in the AI. I'll check it out.

**Comment by Andrettin** [comment]
There was an issue with the AI in Gathering Materials; in Northwards to the Sea I didn't get any issues, though.

In the Hills of the Shorbear Clan quest, the AI is meant to be slow to develop (because each of the enemy players start with one worker while you start with five), since you have two enemies and one of them starts with a fully-built settlement (the Shorbear Clan having many towers with scouts in them). There were two things which were causing the AI to be slower than it should, though: there was a random chance that the Shinsplitter gold deposit was generated in a position that left the workers "locked in" in a space bounded by trees, their mead hall, and the gold deposit. This has been fixed now. Another thing is that workers were preferring to gather gold rocks which were somewhat near their base (but still outside it) than building the gold mine on their gold deposit, and this was making the time they needed to gather resources take longer if the gold rocks were generated in a position that was close enough. I've also fixed this issue so that now the workers make an additional pass checking for very close resources (16-tile radius) before checking for the somewhat close resources (32-tile radius).

Thanks for informing me of this issue :) I released a patch (1.8.3) incorporating the changes I noted.

**Comment by a-detiste** [comment]
> Northwards to the Sea

Yeah, I was just a bit untrained when I first played this one but it's ok for one of the firsts missions.

> Gathering Materials

This one could be easily rushed by running over the enemy with every moving units (including workers) available at first and the let the computer mine coal unattended: not fun. 

> Shorbear Clan 

Yeah, but once all worker are killed, a balista was almost enough to win the game


Will try again. :smile: & have my 12-years old daughter try i tou too.

**Comment by a-detiste** [comment]
>  copper to sometimes be generated instead of gold for the Gathering Materials quest. 

Yup, that was a pain & why  I choosed to rush enemy.

**Comment by Andrettin** [comment]
> This one could be easily rushed by running over the enemy with every moving units (including workers) available at first and then let the computer mine coal unattended: not fun. 

That's a good point. It shouldn't be so easy to destroy the goblin settlements by rushing. I've added three spearmen and two archers for each goblin settlement in normal difficulty (these are used only defensively), with more of them in the higher difficulties.

> Yeah, but once all worker are killed, a balista was almost enough to win the game


> Will try again. :smile: & have my 12-years old daughter try i tou too.

I see what you mean - with the settlement's main defense being the towers and with not many starting units, the player would easily demolish it with a ballista if the Shinsplitters' attacks are deflected. I've added some defensive-only melee troops for the Shorbears defending each tower to prevent that from happening.

(these changes will be in the next patch, which will be released tomorrow)
